### ISSUE TITLE
Add OpenAPI3 specification for ckan aPI

### ISSUE THREAD

**Issue by ioggstream** [ckan/ckan#5160]
### CKAN Version if known (or site URL)

2.6

### Please describe the expected behaviour

- to have an openapi.yaml specification

### Please describe the actual behaviour

- there's no one

### NOTE

Here you can find a superset OAS3 spec for ckan 2.2

https://app.swaggerhub.com/apis/EU-Open-Data-Portal/eu-open_data_portal/0.8.0

**Comment by smotornyuk** [comment]
Good idea. If you have time, please make PR with implementations of this feature. 

Please, note, you cannot use static OpenAPI specification, as any of the installed extensions may change API/response/request/etc., so you need to dynamically determine the structure of API.

**Comment by Gauravp-NEC** [comment]
@smotornyuk , I am interested to work on this issue. Can you help me to understand what is dynamically determine structure of API?

**Comment by smotornyuk** [comment]
CKAN API can be extended. Depending on a list of enabled plugins, every portal has some extra actions. For example, if `datastore` plugin is enabled, new `datastore_create`/`datastore_search` actions are available through API. And API specification must somehow account for these details and there must be a way to extend it every time a new action is registered inside CKAN application.
